src/LabLog/Views/Teacher/Index.cshtml

```diff
@@ -1,9 +1,13 @@
-﻿@{
-    ViewData["Title"] = "List of Rooms";
-}
+﻿@model UserModel
 
-
-<p>This page is the admin landing page.</p>
-<p>It will contain a list of rooms, as well as
-    the options to create a new room. Room editing and deleting will be done from
-    within each Room's view page.</p>
\ No newline at end of file
+  <div class="wrapper">
+    <form class="form-signin" action="/Teacher" method="POST">
+      <h2 class="form-signin-heading">Please login</h2>
+      <input type="text" class="form-control" name="username" placeholder="E-Number" required="True" autofocus="" />
+      <input type="password" class="form-control" name="password" placeholder="Password" required="True"/>      

```

Comment:
there are special generator functions you can use to help with these. [Tag Helpers](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/tag-helpers/intro)

Reply:
Does asp-for make sense in an input field?
`<input asp-for="Email" class="form-control" />`
Example taken from your link.
What is the purpose of asp-for in an input field? It already binds mouseclick to itself.